Go: convert regex-use, url-redirection sinks to use models-as-data

owen-mc opened the pull request:
Convert regex-use, url-redirection sinks to use models-as-data. Follows on from https://github.com/github/codeql/pull/17162.

egregius313 commented on go/ql/lib/semmle/go/frameworks/stdlib/Regexp.qll:
```diff
@@ -6,13 +6,41 @@ import go
+   * function does not compile the regex.
+   */
+  private predicate regexSinkKindInfo(string kind, int strArg) {
+    strArg = -3 and
+    kind = "regex-use"
+    or
+    sinkModel(_, _, _, _, _, _, _, kind, _, _) and
+    exists(string strArgStr |
+      strArgStr.toInt() = strArg
+      or
+      strArg = -2 and
+      strArgStr = "c"
+    |
+      kind = "regex-use[" + strArgStr + "]"
+    )
```
Does this need to add handling of `-1` corresponding to `receiver` like done in the http concepts library?

egregius313 reviewed (changes requested):
Other than my comment about the regex kind, this pretty much LGTM.

However, since this modifies several of the same framework libraries as #17162, I am going to hold off on approving the PR until after the rebasing has happened.

owen-mc commented:
I hadn't realised that the two PRs conflicted. Now that https://github.com/github/codeql/pull/17162 is merged I have rebased (resolving conflicts) and added a commit addressing your review comment.

egregius313 reviewed (approved):
LGTM provided tests pass
